malduck/extractor/extract_manager.py

```diff
@@ -308,13 +308,16 @@ def push_config(self, config: Config, extractor: Extractor) -> None:
         :type config: dict
         :param extractor: Extractor object reference
         :type extractor: :class:`malduck.extractor.Extractor`
+        :param jsonable: Try to decode 'bytes' as UTF-8 and check if config can be converted to JSON (default: True)
+        :type jsonable: bool
         """
-        config = encode_for_json(config)
-        try:
-            json.dumps(config)
-        except (TypeError, OverflowError) as e:
-            log.debug("Config is not JSON-encodable (%s): %s", str(e), repr(config))
-            raise RuntimeError("Config must be JSON-encodable")
+        if jsonable:
+            config = encode_for_json(config)
+            try:
+                json.dumps(config)
+            except (TypeError, OverflowError) as e:
+                log.debug("Config is not JSON-encodable (%s): %s", str(e), repr(config))
+                raise RuntimeError("Config must be JSON-encodable")
 
         config = sanitize_config(config)
```

Comment:
what happens if `jsonable` is false and config was never json-encoded here? Won't this just crash later? :thinking:

Reply:
That `jsonable` field is added because I need to somehow support the `binaries` that contain bytes. CCCS also noted that it's not JSON-compatible https://github.com/CybercentreCanada/Maco/blob/master/maco/model/model.py#L214.

`jsonable` is only checked for configuration part, so it won't crash later in Extractor itself but may crash during further processing of configurations (e.g. `malduck extract` prints JSONs so we need some extra processing there)

`jsonable` sanity check was made to correct mistakes done by extractor writers. But if we follow the Maco scheme, we can trust that type-checking will help in spotting that kind of errors and `bytes` will be only under `binaries` key that can be treated specially.

Reply:
In Assemblyline, we've base64'd any `bytes` using a custom encoder ([Base64Encoder](https://github.com/CybercentreCanada/assemblyline-service-configextractor/blob/2800caa9ad61229a5b80d794d564be30075e74a7/configextractor_/configextractor_.py#L43)) when dropping files to be shown in the analysis.

Not sure if you guys want to go the same route here, but thought I should mention it 😅